scripts/hid_configurator/configurator.py

```diff
@@ -75,16 +75,16 @@ def __repr__(self):
 
     @staticmethod
     def parse_response(response_raw):
-        data_field_len = len(response_raw) - struct.calcsize('HBBB')
+        data_field_len = len(response_raw) - struct.calcsize('<BHBBB')
 
         if data_field_len < 0:
             logging.error('Response too short')
             return None
 
         # Report ID is not included in the feature report from device
-        fmt = '<HBBB{}s'.format(data_field_len)
+        fmt = '<BHBBB{}s'.format(data_field_len)
 
-        (rcpt, event_id, status, data_len, data) = struct.unpack(fmt, response_raw)
+        (report_id, rcpt, event_id, status, data_len, data) = struct.unpack(fmt, response_raw)
```

Comment:
we also need to check if report_id is what we wait for

Reply:
Seems so, especially if we use USB (appllication may set first byte to any chosen value)